File: Gensparky Backend/src/main/java/com/genspark/gensparky/services/UserService.java
package com.genspark.gensparky.services;

import com.genspark.gensparky.entities.User;

import java.util.List;

public interface UserService {
    // get all
    List<User> getUsers();
    // get by id
    User getUserById(Long Id);
    // add
    User adduser(User user);
    // update
    User updateUserById(Long Id, User user);
    // delete by id
    boolean deleteUserById(Long Id);  // returns true if user found and deleted, false if not found
    // get user by discord name
    User getUserByDiscordId(Long discordId);
    // get users on bench
    List<User> getBenchUsers();
    // get trainees in cohorts
    List<User> getTrainees();
}

File: Gensparky Frontend/src/services/UserService.js
import axios from 'axios';

const USERS_REST_API_URL = 'http://24.99.46.153:8081/users';

class UserService {

    getUsers(){
        return axios.get(USERS_REST_API_URL);
    }

    addUser(user){
        return axios.post(USERS_REST_API_URL, user);
    }

    deleteUser(id){
        return axios.delete(USERS_REST_API_URL + `/${id}`);
    }
    
    updateUser(user,id){
        return axios.put(USERS_REST_API_URL + `/${id}`, user);
    }

    getBenchUsers(){
        return axios.get('http://24.99.46.153:8081/users/bench')
    }

    getTrainees(){
        return axios.get('http://24.99.46.153:8081/users/trainees')
    }
}

export default new UserService();